tests/e2e/cleanup_test.go

```diff
@@ -25,53 +25,127 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
-	corev1 "k8s.io/api/core/v1"
+	nfdclient "sigs.k8s.io/node-feature-discovery/api/generated/clientset/versioned"
+	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/api/nfd/v1alpha1"
+
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clientset "k8s.io/client-go/kubernetes"
-	nfdclient "sigs.k8s.io/node-feature-discovery/api/generated/clientset/versioned"
-	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/api/nfd/v1alpha1"
 )
 
-// GetNonControlPlaneNodes gets the nodes that are not tainted for exclusive control-plane usage
-func GetNonControlPlaneNodes(ctx context.Context, cli clientset.Interface) ([]corev1.Node, error) {
-	nodeList, err := cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
-	if err != nil {
-		return nil, err
-	}
-	if len(nodeList.Items) == 0 {
-		return nil, fmt.Errorf("no nodes found in the cluster")
+// cleanupNamespaceResources removes all resources in the specified namespace.
+func cleanupNamespaceResources(namespace string) {
+	err := cleanupTestPods(namespace)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = cleanupHelmDeployments(namespace)
+	Expect(err).NotTo(HaveOccurred())
+
+	cleanupNode(ctx, clientSet)
+	cleanupNFDObjects(ctx, nfdClient, testNamespace.Name)
+	cleanupCRDs()
+}
+
+// waitForDeletion polls the provided checkFunc until a NotFound error is returned,
+// confirming that the resource is deleted.
+func waitForDeletion(resourceName string, checkFunc func() error) error {
+	timeout := 2 * time.Minute
+	interval := 5 * time.Second
+	start := time.Now()
+	for {
+		err := checkFunc()
+		if err != nil && errors.IsNotFound(err) {
+			return nil
+		}
+		if time.Since(start) > timeout {
+			return fmt.Errorf("timed out waiting for deletion of %s", resourceName)
+		}
+		time.Sleep(interval)
 	}
+}
 
-	controlPlaneTaint := corev1.Taint{
-		Effect: corev1.TaintEffectNoSchedule,
-		Key:    "node-role.kubernetes.io/control-plane",
+// cleanupTestPods deletes all test Pods in the namespace that have the label "app.nvidia.com=k8s-dra-driver-gpu-test-app".
+func cleanupTestPods(namespace string) error {
+	labelSelector := "app.nvidia.com=k8s-device-plugin-test-app"
+	podList, err := clientSet.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
```

Comment:
the `ctx` should be passed as a parameter. I see other methods in this PR doing that